kraft-server/src/main/java/com/ozangunalp/kraft/server/BrokerConfig.java

```diff
@@ -69,23 +69,31 @@ public static Properties defaultCoreConfig(Properties props, String host, int ka
             props.put(KafkaConfig.BrokerIdProp(), brokerId);
         }
 
-        // Configure kraft
-        props.putIfAbsent(KafkaConfig.ProcessRolesProp(), "broker,controller");
-        props.putIfAbsent(KafkaConfig.QuorumVotersProp(), brokerId + "@" + controller.host() + ":" + controller.port());
+
+        boolean zookeeper = props.containsKey(KafkaConfig.ZkConnectProp());
+        if (!zookeeper) {
+            // Configure kraft
+            props.putIfAbsent(KafkaConfig.ProcessRolesProp(), "broker,controller");
+            props.putIfAbsent(KafkaConfig.QuorumVotersProp(), brokerId + "@" + controller.host() + ":" + controller.port());
+        }
 
 
-        if (!props.containsKey(KafkaConfig.ControllerListenerNamesProp()) ||
+        if (!(props.containsKey(KafkaConfig.ControllerListenerNamesProp()) || zookeeper) ||
```

Comment:
This block is there to generate listeners config when it is not configured. Maybe with zookeeper we just don't need the controller listeners that's all.